### PR TITLE
ID-931, ID-814 Add test for allowing inactive/disallowed users to access self attributes. Also fixing permissions checking for tests.

### PR DESCRIPTION
**Pull request by Ghost-in-a-Jar**
Ticket: https://broadworkbench.atlassian.net/browse/ID-931

What:

This is a followup to the previous pr https://github.com/broadinstitute/sam/pull/1295 where I fixed a bug. Inactive/disallowed users should be able to still access basic attributes about themselves so the ui can tell whether to serve up the TOS.

Why:

More test coverage!

How:

I noticed that when I made the test, it succeeded even when the endpoint was still wrapped with `withActiveUser`. I realized that the mock SamRoutes these tests were using from MockSamRoutesBuilder replace `withActiveUser` and `withUserAllowInactive` with a no-op. I replaced with the proper mock calls and then had to fix some other tests that were failing. 

  \<If the PR is big, please indicate where a reviewer should start reading it (i.e. which file or function).\>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2Spec.scala [hunk 1]
```diff
@@ -196,6 +196,24 @@ class UserRoutesV2Spec extends AnyFlatSpec with Matchers with ScalatestRouteTest
...
+      .withUserAttributes(defaultUser, userAttributes)
+      .callAsNonAdminUser()
+      .build
+
+    // Act and Assert
+    Get(s"/api/users/v2/self/attributes") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[SamUserAttributes] should be(userAttributes)
+    }
+  }
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2Spec.scala:215]: This is the only net new test

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/service/MockUserServiceBuilder.scala [hunk 2]
```diff
@@ -137,6 +145,9 @@ case class MockUserServiceBuilder() extends IdiomaticMockito {
   }
 
   private def makeUsers(samUsers: Iterable[SamUser], mockUserService: UserService): Unit = {
+    mockUserService.getUserFromGoogleSubjectId(any[GoogleSubjectId], any[SamRequestContext]) answers ((googleSubjectId: GoogleSubjectId) =>
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/service/MockUserServiceBuilder.scala:148]: This is necessary to get the StandardSamUserDirectives.getActiveSamUser and StandardSamUserDirectives.getSamUser calls to work

**Review by Shakespeared — APPROVED** [review]

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/api/AdminServiceUserRoutesSpec.scala [hunk 3]
```diff
@@ -118,6 +118,7 @@ class AdminServiceUserRoutesSpec extends AnyFlatSpec with Matchers with Scalates
     // Arrange
     val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
       .withEnabledUser(defaultUser)
+      .withAllowedUser(defaultUser)
```
Comment (Ghost-in-a-Jar) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/api/AdminServiceUserRoutesSpec.scala:121]: We have a distinction in sam between a user being 'allowed' to use the system (accepted tos for example) and a user being 'enabled' (a flag on the user database model)

Comment (rushtong) [reply src/test/scala/org/broadinstitute/dsde/workbench/sam/api/AdminServiceUserRoutesSpec.scala:121]: This is interesting - I find the phrase `allowed` to be confusing in that an allow state could, in theory, apply to a LOT of different things like `allowed as an admin` or `allowed as a reader` etc. As opposed to something like `withToSAccepted()` or  `withToSRejected()` which is much more specific to the feature you're describing here.

**Review by rushtong — APPROVED** [review]
Looks reasonable to me 👍🏽

**Review by tlangs — APPROVED** [review]

**Pull request merged**